src/text-fragment-utils.js

```diff
@@ -16,7 +16,7 @@
 
 /**
  * @typedef {Object} TextFragment
- * @property {string} textStart
+ * @property {string?} textStart
```

Comment:
I'm pretty sure the spec requires textStart; when would this be missing?

Reply:
+1, textStart is the one required parameter.

Reply:
Sorry, I just double checked and the properties are never null, they are either empty strings or strings with content, removing the optional mark.